### PR TITLE
VACMS-2285 facility services

### PR DESCRIPTION
**Pull request by ethanteague**
## Description
See #2285 

## Testing done
Visual, behat

## Screenshots
image
image

## QA steps
- As an admin, go to `/node/350/edit` and visually verify that you see an `Add service location` button
- Click on it, and confirm it brings up three fieldsets: Address, phone, hours
- Enter data into each fieldset, create a second instance of service location, and enter data in it.
- Save page, and visually verify that all data is available.

### CODE REVIEW

File: config/sync/core.entity_form_display.node.va_form.default.yml [hunk 1]
```diff
@@ -181,19 +177,6 @@ content:
...
-      display_label: true
-    third_party_settings: {  }
-    type: boolean_checkbox
-    region: content
-  field_va_form_deleted_date:
-    weight: 26
-    settings: {  }
-    third_party_settings: {  }
-    type: datetime_default
-    region: content
```
Comment (swirtSJW) [inline config/sync/core.entity_form_display.node.va_form.default.yml:196]: Major:  Should not be going away

File: config/sync/core.entity_form_display.node.va_form.default.yml [hunk 2]
```diff
@@ -104,8 +103,6 @@ third_party_settings:
         - field_va_form_revision_date
         - field_va_form_administration
         - field_va_form_num_pages
-        - field_va_form_deleted
-        - field_va_form_deleted_date
```
Comment (swirtSJW) [inline config/sync/core.entity_form_display.node.va_form.default.yml:108]: Major: These should not be going away.   (config or db mismatch)  They just got merged in yesterday.

File: config/sync/core.entity_view_display.node.va_form.default.yml [hunk 3]
```diff
@@ -8,8 +8,6 @@ dependencies:
     - field.field.node.va_form.field_benefit_categories
     - field.field.node.va_form.field_meta_tags
     - field.field.node.va_form.field_va_form_administration
-    - field.field.node.va_form.field_va_form_deleted
-    - field.field.node.va_form.field_va_form_deleted_date
```
Comment (swirtSJW) [inline config/sync/core.entity_view_display.node.va_form.default.yml:12]: Major: Should be staying

File: config/sync/core.entity_view_display.node.va_form.default.yml [hunk 4]
```diff
@@ -63,8 +61,6 @@ third_party_settings:
     group_from_forms_db:
       children:
         - field_va_form_administration
-        - field_va_form_deleted
-        - field_va_form_deleted_date
```
Comment (swirtSJW) [inline config/sync/core.entity_view_display.node.va_form.default.yml:67]: staying

File: config/sync/core.entity_view_display.node.va_form.default.yml [hunk 5]
```diff
@@ -176,8 +172,6 @@ hidden:
   content_moderation_control: true
   field_administration: true
   field_va_form_administration: true
-  field_va_form_deleted: true
-  field_va_form_deleted_date: true
```
Comment (swirtSJW) [inline config/sync/core.entity_view_display.node.va_form.default.yml:180]: staying

File: config/sync/core.entity_view_display.node.va_form.external_content.yml [hunk 6]
```diff
@@ -9,8 +9,6 @@ dependencies:
     - field.field.node.va_form.field_benefit_categories
     - field.field.node.va_form.field_meta_tags
     - field.field.node.va_form.field_va_form_administration
-    - field.field.node.va_form.field_va_form_deleted
-    - field.field.node.va_form.field_va_form_deleted_date
```
Comment (swirtSJW) [inline config/sync/core.entity_view_display.node.va_form.external_content.yml:13]: staying

File: config/sync/core.entity_view_display.node.va_form.external_content.yml [hunk 7]
```diff
@@ -75,25 +73,6 @@ content:
...
-    third_party_settings: {  }
-  field_va_form_deleted_date:
-    type: datetime_default
-    weight: 8
-    region: content
-    label: above
-    settings:
-      timezone_override: ''
-      format_type: short_date_no_time
-    third_party_settings: {  }
```
Comment (swirtSJW) [inline config/sync/core.entity_view_display.node.va_form.external_content.yml:96]: stayinhg

File: config/sync/core.entity_view_display.node.va_form.external_content.yml [hunk 8]
```diff
@@ -155,6 +155,8 @@ hidden:
   field_alert: true
   field_benefit_categories: true
   field_meta_tags: true
+  field_va_form_deleted: true
+  field_va_form_deleted_date: true
```
Comment (swirtSJW) [inline config/sync/core.entity_view_display.node.va_form.external_content.yml:159]: Major: These should not have moved to hidden.  They are supposed to be visible in this display.

File: config/sync/migrate_plus.migration.va_node_form.yml [hunk 9]
```diff
@@ -253,6 +253,9 @@ process:
   sticky:
     plugin: default_value
     default_value: 0
+  status:
+    plugin: default_value
+    default_value: 0
```
Comment (swirtSJW) [inline config/sync/migrate_plus.migration.va_node_form.yml:258]: Major: Why is this here?   Nothing about your PR should be touching va_form.  You have a bad export... makes me think something is out of date and puts other changes in suspect.

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module [hunk 10]
```diff
@@ -378,8 +377,14 @@ function va_gov_backend_field_widget_paragraphs_form_alter(&$element, FormStateI
     }
   }
 
-  if (in_array($paragraph_entity_reference_field_name, ['field_phone_numbers_paragraph', 'field_appointment_phone_numbers'])) {
-    _va_gov_backend_modify_phone_paragraph_interactions($element, $form_state, $context);
+  // If we have a field that refs certain paragraphs, fire function.
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_backend/va_gov_backend.module:380]: Minor: Can you be more specific about these "certain paragraphs" that makes them subject to the special handling.  We want the rational to persist.

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module [hunk 11]
```diff
@@ -378,8 +377,14 @@ function va_gov_backend_field_widget_paragraphs_form_alter(&$element, FormStateI
...
+  if (in_array($paragraph_entity_reference_field_name, [
+    'field_service_location',
+    'field_service_location_address',
+    'field_phone_numbers_paragraph',
+    'field_phone',
+  ])) {
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_backend/va_gov_backend.module:386]: Minor: I'm not a fain of the readability of this pattern composing the array in the function.  
I'd prefer to see them broken out into a meaningfully named array,.
```suggestion
  $vha_paragraphs = [
    'field_service_location',
    'field_service_location_address',
    'field_phone_numbers_paragraph',
    'field_phone',
  ];
  if (in_array($paragraph_entity_reference_field_name, $vha_paragraphs)) {
```

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module [hunk 12]
```diff
@@ -393,7 +398,7 @@ function va_gov_backend_field_widget_paragraphs_form_alter(&$element, FormStateI
  * @param array $context
  *   The context.
  */
-function _va_gov_backend_modify_phone_paragraph_interactions(array &$element, FormStateInterface &$form_state, array &$context) {
+function _va_gov_backend_modify_paragraph_interactions(array &$element, FormStateInterface &$form_state, array &$context, $paragraph_entity_reference_field_name) {
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_backend/va_gov_backend.module:401]: Good renaming to make it more accurate.  

Minor: It looks like the function description needs updating.
Major:  Missing an @param for the new param.

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module [hunk 13]
```diff
@@ -404,20 +409,63 @@ function _va_gov_backend_modify_phone_paragraph_interactions(array &$element, Fo
   $paragraph_instance = $widget_state['paragraphs'][$element['#delta']]['entity'];
   $paragraph_type = $paragraph_instance->bundle();
 
+  // Toggle hours fields depending on "Use facility hours" select.
+  // field_service_location[0][subform][field_service_location_address][0][subform][field_use_facility_address][value].
+  if (in_array($paragraph_type, [
+    'service_location',
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_backend/va_gov_backend.module:415]: Is there  a reason fo an array of 1?   Are more coming?  If so, lets break it out so the array is a variable that describes what it contains.

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module [hunk 14]
```diff
@@ -404,20 +409,63 @@ function _va_gov_backend_modify_phone_paragraph_interactions(array &$element, Fo
...
+  ])) {
+    $toggle_fields = [
+      'field_facility_service_hours',
+    ];
+    $selector = sprintf(':input[name="field_service_location[%1$d][subform][field_hours]"]', $element['#delta']);
+    foreach ($toggle_fields as $field) {
+      $element['subform'][$field]['#states'] = [
+        'visible' => [
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_backend/va_gov_backend.module:423]: Minor:  Comment on what this is doing and more importantly why.

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module [hunk 15]
```diff
@@ -404,20 +409,63 @@ function _va_gov_backend_modify_phone_paragraph_interactions(array &$element, Fo
...
+          $selector => ['value' => "2"],
+        ],
+      ];
+    }
+  }
+
+  // Toggle address field depending on "Use facility address?" boolean.
+  if (in_array($paragraph_type, [
+    'service_location_address',
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_backend/va_gov_backend.module:432]: single item arrays?

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module [hunk 16]
```diff
@@ -404,20 +409,63 @@ function _va_gov_backend_modify_phone_paragraph_interactions(array &$element, Fo
...
+        ],
+      ];
+    }
+  }
+
   // 1. Hide the extension field if phone option is not selected.
   // 2. Add constraint error message jumplink to phone field.
-  if ($paragraph_type === 'phone_number') {
+  if (in_array($paragraph_type, [
+    'phone_number',
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_backend/va_gov_backend.module:453]: I see you using this pattern of single item arrays inline.   Maybe I am missing the reason.  Can you clarify?  I find it to be less meaningful and less readable.

File: tests/behat/drupal-spec-tool/content_model_bundles.feature [hunk 17]
```diff
@@ -5,65 +5,68 @@ Feature: Content model bundles
   I want to have content entity bundles that reflect my content model.
 
   @dst @content_type @dstbundles
-  Scenario: Bundles
```
Comment (swirtSJW) [inline tests/behat/drupal-spec-tool/content_model_bundles.feature:8]: copy and paste seems to have changed the format which makes them look like they all changed.  Can you adjust so that only the lines that are changed, changed?

File: tests/behat/drupal-spec-tool/content_model_content_type_fields.feature [hunk 18]
```diff
@@ -5,286 +5,288 @@ Feature: Content model: Content Type fields
   I want to have content type fields that reflect my content model.
 
   @dst @field_type @content_type_fields @dstfields
-  Scenario: Fields
```
Comment (swirtSJW) [inline tests/behat/drupal-spec-tool/content_model_content_type_fields.feature:8]: formatting change?

File: tests/behat/drupal-spec-tool/content_model_paragraph_fields.feature [hunk 19]
```diff
@@ -5,60 +5,72 @@ Feature: Content model: Paragraph fields
   I want to have paragraph fields that reflect my content model.
 
   @dst @field_type @paragraph_fields @dstfields
-  Scenario: Fields
```
Comment (swirtSJW) [inline tests/behat/drupal-spec-tool/content_model_paragraph_fields.feature:8]: formatting changes?

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module (hunk 16 above)
Comment (ethanteague) [reply docroot/modules/custom/va_gov_backend/va_gov_backend.module:453]: My thinking is that we might be adding new paragraph types as our codebase grows to this function.....but you are right - unnecessary / lead readable at this time, @swirtSJW

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module (hunk 12 above)
Comment (ethanteague) [reply docroot/modules/custom/va_gov_backend/va_gov_backend.module:401]: My bad - code editor didn't give me that red squiggle :+1:

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module (hunk 13 above)
Comment (ethanteague) [reply docroot/modules/custom/va_gov_backend/va_gov_backend.module:427]: Probably not - I'll get rid of this array pattern.

File: tests/behat/drupal-spec-tool/content_model_bundles.feature [hunk 20]
```diff
@@ -16,9 +16,9 @@ Feature: Content model bundles
       | Alert | alert | Paragraph type | A reusable or non-reusable alert, either "information status" or "warning status". |
       | Benefits detail page | page | Content type | These pages hold all of the benefits overview content, such the detail pages linked to from va.gov/disability, va.gov/health-care, and va.gov/education. |
       | Benefits hub landing page | landing_page | Content type | A special page for top-level Benefits content with its own one-off layout and content. |
-      | CMS help page | documentation_page | Content type | Help pages for VA.gov CMS editors. |
       | Detail Page | health_care_region_detail_page | Content type | For static pages where there's not another content type already available.  |
       | Document | document | Media type | A locally hosted document, such as a PDF. |
+      | CMS help page | documentation_page | Content type | Help pages for VA.gov CMS editors. |
```
Comment (swirtSJW) [inline tests/behat/drupal-spec-tool/content_model_bundles.feature:21]: Bad alpha sort?

File: tests/behat/drupal-spec-tool/content_model_content_type_fields.feature [hunk 21]
```diff
@@ -137,7 +137,7 @@ Feature: Content model: Content Type fields
       | Content type | Publication listing page | Meta title tag | field_meta_title | Text (plain) | Required | 1 | Textfield with counter | Translatable |
       | Content type | Publication listing page | Owner | field_administration | Entity reference | Required | 1 | Select list | Translatable |
       | Content type | Publication listing page | Related office or health care system | field_office | Entity reference | Required | 1 | Select list | Translatable |
-      | Content type | Q&A | Answer | field_answer | Entity reference revisions | Required | Unlimited | Paragraphs EXPERIMENTAL | |
+      | Content type | Q&A | Answer | field_answer | Entity reference revisions | Required | Unlimited | Paragraphs EXPERIMENTAL |  |
```
Comment (swirtSJW) [inline tests/behat/drupal-spec-tool/content_model_content_type_fields.feature:140]: Adding a space?

**Comment by swirtSJW** [comment]
Seeing bizarre behat failures due to 
outreach 

image

and 

image

**Comment by ethanteague** [comment]
> Seeing bizarre behat failures due to
> outreach asset
> 
> image
> 
> and
> 
> image

Seeing a lot of these, locally: ```Uncaught PHP Exception PHPUnit\Framework\Error\Notice: "Undefined index: HTTP_HOST" at /app/docroot/vendor/simplesamlphp/simplesamlphp/config/config.php line 30```
